Executing tests

Hi,

do you have an idea, how to package pybind11_tests, that your tests require?

Pete

After switching my build to git master, I'm facing another **strange** issue. All tests but exactly one succeed:
```
[   32s] + PYTHONPATH=./build/lib.linux-x86_64-3.8
[   32s] + python3 -m pytest tests
[   32s] ============================= test session starts ==============================
[   32s] platform linux -- Python 3.8.5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
[   32s] rootdir: /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1+git20200930.d09bfc7
[   32s] collected 172 items
[   32s] 
[   33s] tests/test_destination.py ...                                            [  1%]
[   33s] tests/test_document.py ....F............................................ [ 30%]
[   33s] ..............................................                           [ 56%]
[   33s] tests/test_font.py ....                                                  [ 59%]
[   33s] tests/test_image.py .....                                                [ 62%]
[   33s] tests/test_page.py ...............                                       [ 70%]
[   33s] tests/test_pagerenderer.py ............                                  [ 77%]
[   33s] tests/test_pagetransition.py ..........................                  [ 93%]
[   33s] tests/test_rectangle.py ......                                           [ 96%]
[   33s] tests/test_toc.py ...                                                    [ 98%]
[   33s] tests/test_version.py ...                                                [100%]
[   33s] 
[   33s] =================================== FAILURES ===================================
[   33s] _____________________________ test_load_with_file ______________________________
[   33s] 
[   33s] data_path = PosixPath('/home/abuild/rpmbuild/BUILD/python-poppler-0.2.1+git20200930.d09bfc7/tests/data')
[   33s] 
[   33s]     def test_load_with_file(data_path):
[   33s]         with (data_path / "document.pdf").open("rb") as f:
[   33s]             pdf_document = document.load(f, "owner", "user")
[   33s]         if version() < (0, 46, 0):
[   33s]             assert pdf_document.author == "Charles"
[   33s]         else:
[   33s] >           assert pdf_document.author == "Charles Brunet"
[   33s] E           AssertionError: assert '' == 'Charles Brunet'
[   33s] E             - Charles Brunet
[   33s] 
[   33s] tests/test_document.py:73: AssertionError
[   33s] ----------------------------- Captured stderr call -----------------------------
[   33s] poppler/error (7622): Command token too long
[   33s] poppler/error (7750): Command token too long
[   33s] poppler/error (7878): Command token too long
[   33s] poppler/error (8006): Command token too long
[   33s] poppler/error (8134): Command token too long
[   33s] poppler/error: xref num 19 not found but needed, try to reconstruct<0a>
[   33s] poppler/error: Couldn't find trailer dictionary
[   33s] =========================== short test summary info ============================
[   33s] FAILED tests/test_document.py::test_load_with_file - AssertionError: assert '...
[   33s] ======================== 1 failed, 171 passed in 0.44s =========================
```

Running the same code from a git checkout does work properly. Puzzled. Given, that all the variations do work well, but the file argument does not, just don't make any sense to me.. Does this ring a bell for you?

--- a/NEWS.txt
+++ b/NEWS.txt
@@ -1,5 +1,8 @@
 Changelog
 
+0.2.2
+- Bugfix: Document can be corrupted by garbage collector when created from data (or from file handle) (fixes #19)
+
 0.2.1
 - Bugfix: Fixed byte order in image format string used for Pillow
 - Bugfix: Prevent segmentation fault when performing some operations on locked documents (fixes #4)

--- a/src/poppler/document.py
+++ b/src/poppler/document.py
@@ -53,8 +53,9 @@ class Document:
     PageLayout = document.page_layout_enum
     PageMode = document.page_mode_enum
 
-    def __init__(self, poppler_document):
+    def __init__(self, poppler_document, data=None):
         self._document = poppler_document
+        self._data = data
 
     @ensure_unlocked
     def create_font_iterator(self, page=0):
@@ -282,7 +283,8 @@ def load_from_file(file_name, owner_password=None, user_password=None):
 
 def load_from_data(file_data: bytes, owner_password=None, user_password=None):
     return Document(
-        document.load_from_data(file_data, owner_password or "", user_password or "")
+        document.load_from_data(file_data, owner_password or "", user_password or ""),
+        file_data
     )
 
 

